ci/scripts/run_smoke_test.js

```diff
@@ -28,14 +33,65 @@ async function getS3Objects (splitLocation, prefix) {
   return s3.listObjects({ Bucket: splitLocation[0], Prefix: prefix }).promise()
 }
 
+function checkIfCanaryIsInStartableState (canary) {
+  const canaryStatus = canary.Canary.Status.State
+  switch (canaryStatus) {
+    case 'CREATING':
+    case 'STARTED':
+    case 'RUNNING':
+    case 'UPDATING':
+    case 'STOPPING':
+      return 'wait'
+    case 'STOPPED':
+      console.log('Canary is ready to start')
+      return 'ready'
+    case 'DELETING':
+    case 'ERROR':
+      throw new Error(`Unable to start canary in state ${canaryStatus}`)
+  }
+}
+
+function sleep (ms) {
+  return new Promise(resolve => setTimeout(resolve, ms))
+}
+
 async function run () {
+  // Assume the canary is ready for now
+  let canaryStatus = 'ready'
+
+  try {
+    const canary = await getCanary()
+    canaryStatus = checkIfCanaryIsInStartableState(canary)
+  } catch (error) {
+    console.log(error)
+    process.exit(1)
+  }
+
+  if (canaryStatus === 'wait') {
+    // Wait 60 secs and continue
+    console.log('Canary is not in a startable state. Wait 60s before starting.')
+    try {
+      await sleep(CANARY_TIMEOUT)
+    } catch (error) {
+      console.log(error)
+    }
```

Comment:
not really sure there'll be an error here but not a biggie

Reply:
Yeah, it's probably overkill but we had a few warnings pop up about uncaught promises, so wanted to make sure it wasn't this one.